### PR TITLE
Bookmarks - Display for user episodes

### PR DESCRIPTION
**Pull request by ashiagr**
## Description

This displays bookmarks for user episodes (files).

There's no Figma design, but here's the iOS PR: https://github.com/Automattic/pocket-casts-ios/pull/1018

## Testing Instructions
- Open a user episode (files)
- Play it and add a few bookmarks
- Return to the Files screen
- ✅ Notice that the file has a bookmark icon
- Tap on the file row
- ✅ Notice that the file details bottom sheet has a bookmark menu
- Tap on the bookmarks menu
- ✅ Notice that the bookmarks dialog is shown with bookmarks for the selected file

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/8084a6a6-7017-4593-b83f-654b589037ed

#### Known Issues:
- Theming doesn't work well
- On deleting all bookmarks for a file, the bookmark icon is not removed for it until the view is reloaded
- Reuses sort order previously set for episodes 

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

File: modules/features/player/src/main/res/layout/fragment_bookmarks_container.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            android:layout_width="48dp"
+            android:layout_height="48dp"
+            android:background="?attr/selectableItemBackgroundBorderless"
+            android:contentDescription="@string/close"
+            android:paddingStart="16dp"
+            android:paddingEnd="16dp"
+            android:src="@drawable/ic_chevron"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:tint="?attr/primary_icon_02" />
```
Comment (geekygecko) [inline modules/features/player/src/main/res/layout/fragment_bookmarks_container.xml:23]: Changing these colors might help a little.

```suggestion
            app:tint="?attr/secondary_icon_01" />
```

| Before | After |
| --- | --- |
| Screenshot_20230807_211109 | Screenshot_20230807_214617 |
| Screenshot_20230807_214145 | Screenshot_20230807_214515 |

File: modules/features/player/src/main/res/layout/fragment_bookmarks_container.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            android:id="@+id/title"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/bookmarks"
+            style="@style/H30"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="@id/btnClose"
+            app:layout_constraintBottom_toBottomOf="@id/btnClose"
+            app:layout_constraintEnd_toEndOf="parent"
+            android:textColor="?attr/primary_text_02" />
```
Comment (geekygecko) [inline modules/features/player/src/main/res/layout/fragment_bookmarks_container.xml:35]:
```suggestion
            android:textColor="?attr/secondary_text_01" />
```

File: modules/features/player/src/main/res/layout/fragment_bookmarks_container.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (geekygecko) [inline modules/features/player/src/main/res/layout/fragment_bookmarks_container.xml:2]: This needs a scroll view as you can't access the bookmarks if they go off the screen.

**Review by geekygecko — APPROVED** [review]

**Comment by ashiagr** [comment]
> This needs a scroll view as you can't access the bookmarks if they go off the screen.

Fixed in https://github.com/Automattic/pocket-casts-android/pull/1239/commits/dc83bc247f2ea21814d3aef12937c3279bd0d3b7

https://github.com/Automattic/pocket-casts-android/assets/1405144/6417ba23-04c7-46b6-b24b-0440041805b6

**Comment by ashiagr** [comment]
👋 @CookieyedCodes 

> As this is the first time I am seeing the player with bookmarks (its only saves play points correct not segments of audio?)

That's correct, saves only play points.

> it might be a tad nice the the player scrub bar places lines or pins or points at that point (ie some UI signification) on where the bookmarks are located, I would also then recommend an icon somewhere in the player or in settings to enable and disable snapping to the bookmark points 🤔🤔(prefer the former)

Let's think of current changes as v1 to enable bookmarking and as a base for incremental improvements. Thanks for all the suggestions. 🙏

**Comment by CookieyedCodes** [comment]
No bothers as usual it's the little things 😉 ❤️

**Pull request merged**